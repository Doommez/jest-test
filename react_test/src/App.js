import { useEffect, useState } from "react";
import "./App.css";

function App() {
  const [data, setData] = useState(null);

  const [toggle, setToggle] = useState(false);

  const [value, setValue] = useState("");

  const onClick = () => {
    setToggle((prev) => !prev);
  };

  const onChange = (e) => {
    setValue(e.target.value);
  };
  useEffect(() => {
    const timout = setTimeout(() => {
      setData("kkk");
    }, 100);

    return () => {
      clearTimeout(timout);
    };
  }, []);
  return (
    <div className="App">
      {toggle == true && <div data-testid={"toggle-elem"}>toggle</div>}
      <h1>Hello world</h1>
      <p>{data ? data : "loading..."}</p>
      <input
        value={value}
        onChange={onChange}
        type="text"
        placeholder="input"
      />
      <button data-testid={"toggle-btn"} onClick={onClick}>
        click me
      </button>
    </div>
  );
}

export default App;
