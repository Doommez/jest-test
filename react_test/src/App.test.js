import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import { toBeInTheDocument } from "@testing-library/jest-dom/matchers";
import { MemoryRouter } from "react-router-dom";

describe("App", () => {
  test("link", () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });
  // test("link", () => {
  //   render(<App />);
  //   const HelloWorldElemnt = screen.getByText(/Hello world/i);
  //   const btn = screen.getByRole("button");
  //   const input = screen.getByPlaceholderText(/input/i);
  //   expect(HelloWorldElemnt).toBeInTheDocument();
  //   expect(btn).toBeInTheDocument();
  //   expect(input).toMatchSnapshot();
  // });
  // test("second", async () => {
  //   render(<App />);
  //   const HelloWorldElemnt = await screen.findByText(/kkk/i);
  //   expect(HelloWorldElemnt).toBeInTheDocument();
  // });
  // test("toggle", () => {
  //   render(<App />);
  //   const btn = screen.getByTestId("toggle-btn");
  //   const div = screen.queryByTestId("toggle-elem");
  //   expect(div).toBeNull();
  //   fireEvent.click(btn);
  //   expect(screen.getByTestId("toggle-elem")).toBeInTheDocument();
  // });
  // test("input", () => {
  //   render(<App />);
  //   const input = screen.getByPlaceholderText(/input/i);
  //   expect(input).toContainHTML("");
  //   fireEvent.input(input, {
  //     target: {
  //       value: "12341234",
  //     },
  //   });
  //   expect(input).toContainHTML("12341234");
  // });
});
