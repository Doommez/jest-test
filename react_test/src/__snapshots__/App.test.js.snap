// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App link 1`] = `
<input
  placeholder="input"
  type="text"
  value=""
/>
`;
